### PR TITLE
Fail on any exit code if configured to fail

### PR DESCRIPTION
**Pull request by pbrisbin**
I recommend by-commit review. The initial commits (listed here) are clear in their intent,
but somewhat broken. Commits that fix things come after.

---

### [Fail on any exit code if configured to fail](https://github.com/freckle/weeder-action/pull/18/commits/e313217db6b7a36532b75f97d4dd4fdc9700a687)
[e313217](https://github.com/freckle/weeder-action/pull/18/commits/e313217db6b7a36532b75f97d4dd4fdc9700a687)

And don't fail regardless of exit code if not configured to fail. This
replaces the previous behavior of ignoring exit code and look for output
in the log to determine if weeds were found or not.

Before, a fatal error that produced no stdout, would not fail. This is
much better. The only gap it leaves is if we're configured _not_ to
fail, we probably still want fatal errors other than weeds-found to fail
the Job. We will resolve that next.

Fixes https://github.com/freckle/weeder-action/issues/13.

### [Fail on fatal errors with fail:false if on v2.7](https://github.com/freckle/weeder-action/pull/18/commits/f21083b160f9a73c084b4d462c675d2cfcffbdee)
[f21083b](https://github.com/freckle/weeder-action/pull/18/commits/f21083b160f9a73c084b4d462c675d2cfcffbdee)

If we're using weeder v2.7 we can trust the exit code to know that 288
means weeds found. Any other exit code should still fail the Job, even
if configured with `fail: false`.

### [Rev actions versions](https://github.com/freckle/weeder-action/pull/18/commits/70605fc139110faec3770c662d2984d661775879)
[70605fc](https://github.com/freckle/weeder-action/pull/18/commits/70605fc139110faec3770c662d2984d661775879)

### [Add 9.6 example, use matrix](https://github.com/freckle/weeder-action/pull/18/commits/11f7e0eb4cc4b09c274a61f6b5b21ef68621752e)

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -31,9 +31,27 @@ runs:
...
         url: "https://github.com/freckle/weeder-action/releases/download/Binaries/{name}-{version}-{os}-{arch}.{ext}"
 
+    - id: weeder-version
+      name: Record weeder version
+      shell: bash
+      run: |
+        {
+          printf 'result='
+          weeder --version | sed '/^weeder version \(.*\)$/!d; s//\1/'
+        } >>"$GITHUB_OUTPUT"
```
Comment (pbrisbin) [inline action.yml:41]:
```suggestion
        weeder --version | sed '/^weeder version \(.*\)$/!d; s//result=\1/' >>"$GITHUB_OUTPUT"
```

File: action.yml [hunk 2]
```diff
@@ -31,9 +31,27 @@ runs:
...
+
+    - id: weeder-check
+      name: Compare weeder version with v2.7
+      uses: pbrisbin/semver-compare-action@master
+      with:
+        left-semver: ${{ steps.weeder-version.outputs.result }}
+        right-semver: 2.7.0
```
Comment (pbrisbin) [inline action.yml:48]:
```suggestion
        right-semver: 2.7.0 # version at which known exit codes were introduced
```

File: action.yml [hunk 3]
```diff
@@ -31,9 +31,24 @@ runs:
...
 
+    - id: weeder-version
+      name: Record weeder version
+      shell: bash
+      run: |
+        weeder --version | sed '/^weeder version \(.*\)$/!d; s//result=\1/' >>"$GITHUB_OUTPUT"
+
+    - id: weeder-check
+      name: Compare weeder version with v2.7
+      uses: pbrisbin/semver-compare-action@master
```
Comment (stackptr) [inline action.yml:42]: Looks like this fork just removes warnings about an outdated version of Node?

**Review by stackptr — APPROVED** [review]
The shell script gets a bit beyond me but the tests look comprehensive.

File: action.yml (hunk 3 above)
Comment (pbrisbin) [reply action.yml:42]: It does. The PR sits open with no response for weeks now though, so I might be just taking over for myself...

**Pull request merged**